src/logging.rs

```diff
@@ -0,0 +1,119 @@
+use std::fs::{create_dir_all, OpenOptions};
+use std::path::{Path, PathBuf};
+
+use anyhow::{Context, Result};
+use tracing::info;
+use tracing::metadata::LevelFilter;
+use tracing_appender::non_blocking;
+use tracing_appender::non_blocking::{NonBlocking, WorkerGuard};
+use tracing_subscriber::reload;
+use tracing_subscriber::{
+    fmt, prelude::__tracing_subscriber_SubscriberExt, util::SubscriberInitExt, EnvFilter, Layer,
+};
+
+type ChangeLogWriter = Box<dyn Fn(NonBlocking) -> Result<()>>;
+
+pub struct DynamicLogManager {
+    /// Flushes the logs when it's dropped.
+    guard: WorkerGuard,
+    log_dir: Option<PathBuf>,
+    change_log_writer: ChangeLogWriter,
+}
+
+impl DynamicLogManager {
+    #[allow(dead_code)]
```

Comment:
Could you explain more on why this is needed?

Reply:
This method is not called in this PR but will be used in a different issue (#113).
As I understand, this PR should only include code related to changing the log directory.
If this method were used here, it would also involve code for fetching remote configuration, adding unnecessary complexity.

Reply:
The plan to use this method later should be commented alongside this allowance. This can serve as a reminder to remove it after completing the task.

Reply:
Applied the suggested adjustment.